### PR TITLE
feat: [CDS-44537]: Adding terraform resource for manual deployment freeze

### PR DESCRIPTION
**Pull request by Y0709**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: examples/resources/harness_platform_manual_freeze/import.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Import freeze
+terraform import harness_platform_manual_freeze.example <org_id>/<project_id>/<freeze_id>
```
Comment (rathodmeetsatish) [inline examples/resources/harness_platform_manual_freeze/import.sh:2]: If its a multilevel entity please add examples for all levels refer any connector's import file

File: internal/service/platform/manual_freeze/resource_manual_freeze.go [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+	id := d.Get("identifier").(string)
+
+	resp, httpResp, err := c.FreezeCRUDApi.GetFreeze(ctx, c.AccountId, id, &nextgen.FreezeCRUDApiGetFreezeOpts{
+		OrgIdentifier:     helpers.BuildField(d, "org_id"),
+		ProjectIdentifier: helpers.BuildField(d, "project_id"),
+	})
+
+	if err != nil {
+		return helpers.HandleApiError(err, d, httpResp)
```
Comment (rathodmeetsatish) [inline internal/service/platform/manual_freeze/resource_manual_freeze.go:175]: use HandleReadApiError here

File: internal/service/platform/manual_freeze/resource_manual_freeze_test.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+	"testing"
+
+	"github.com/antihax/optional"
+	"github.com/harness/harness-go-sdk/harness/nextgen"
+	"github.com/harness/harness-go-sdk/harness/utils"
+	"github.com/harness/terraform-provider-harness/internal/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
```
Comment (rathodmeetsatish) [inline internal/service/platform/manual_freeze/resource_manual_freeze_test.go:15]: add deleteunderlying resource test refer TestAccResourceProject_DeleteUnderlyingResource test

File: examples/resources/harness_platform_manual_freeze/import.sh (hunk 1 above)
Comment (Y0709) [reply examples/resources/harness_platform_manual_freeze/import.sh:8]: done

File: internal/service/platform/manual_freeze/resource_manual_freeze.go (hunk 2 above)
Comment (Y0709) [reply internal/service/platform/manual_freeze/resource_manual_freeze.go:181]: Not needed as type cast was not getting handled properly so added new resource part seperatly

File: internal/service/platform/manual_freeze/resource_manual_freeze_test.go (hunk 3 above)
Comment (Y0709) [reply internal/service/platform/manual_freeze/resource_manual_freeze_test.go:16]: done

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**